libdrgn/symbol.c

```diff
@@ -13,6 +14,36 @@ LIBDRGN_PUBLIC void drgn_symbol_destroy(struct drgn_symbol *sym)
 	free(sym);
 }
 
+LIBDRGN_PUBLIC void
+drgn_symbol_init_from_elf(const char *name, uint64_t address,
+			  GElf_Sym elf_sym, struct drgn_symbol *ret)
+{
+	int binding, kind;
+	ret->name = name;
+	ret->address = address;
+	ret->size = elf_sym.st_size;
+	binding = GELF_ST_BIND(elf_sym.st_info);
+	if (binding < 0 || binding >= STB_NUM) {
+		if (binding == STB_GNU_UNIQUE)
+			ret->binding = DRGN_SYMBOL_BINDING_UNIQUE;
+		else
+			ret->binding = DRGN_SYMBOL_BINDING_UNKNOWN;
+	} else {
+		/* offset by 1 from elf.h definitions */
+		ret->binding = binding + 1;
+	}
```

Comment:
We should compare against the largest constant that we actually know about, just in case `elf.h` gets new constants and raises `STB_NUM`.

`ST_BIND` is defined as `(unsigned char)st_info >> 4`, so binding will always be non-negative. We can simplify it a bit further:

```suggestion
	if (binding <= STB_WEAK || binding == STB_GNU_UNIQUE)
		ret->binding = binding + 1;
	else
		ret->binding = DRGN_SYMBOL_BINDING_UNKNOWN;
```